src/ShallowWaterExplicit.jl

```diff
@@ -0,0 +1,171 @@
+function shallow_water_explicit_time_step!(model, dΩ, dω, f, g, h₁, u₁, hₘ, uₘ, hₚ, uₚ, RTMMchol, L2MMchol, dt, leap_frog, τ, Q, V, R, S, h₂, u₂, ϕ, F)
```

Comment:
Let us finish this round of optimization first, but I would like that we give some more profound thinking towards homogeneization and satisfaction of coding guidelines to the signatures of the different functions that we write in a third round. Things that come into my mind:

- Those arguments which are modified by side-effecting functions (i.e., marked by `!`  by convention) are usually placed first. 
- Can we reduce the number of arguments? E.g., `g` is a constant defined in `GridapGeosciences.jl`, I think there is no need to pass it. 
- All finite element functions corresponding to the different time steps can be passed as a tuple. 
-  Why do we need `R`, and not `P` and `U`. All FE spaces can be passed also as a tuple. 
- All cholesky factorizations/solvers needed can be also passed as a tuple.
- What else?

Reply:
I have not removed `g` as there may be scenarios where we want to override the default value (three that come to mind are non-dimensionalised configurations, exoplanets (with a different `g`) and deep atmosphere approximations (where `g` can vary with radius). Hope that's OK

Also I haven't packaged everything up into tuples as not all the FESpaces or Cholesky factorisations are needed by the time stepper, and I'm wondering if there a performance cost to passing unneeded ones in. That said if you think it is better I'm happy to make a tuple out of the ones that are needed and pass those in....

Reply:
> I have not removed g as there may be scenarios where we want to override the default value (three that come to mind are non-dimensionalised configurations, exoplanets (with a different g) and deep atmosphere approximations (where g can vary with radius). Hope that's OK

Ok, that makes sense. 

> Also I haven't packaged everything up into tuples as not all the FESpaces or Cholesky factorisations are needed by the time stepper, and I'm wondering if there a performance cost to passing unneeded ones in. That said if you think it is better I'm happy to make a tuple out of the ones that are needed and pass those in....

It's ok, I am concerned about the very large number of arguments. But the tuple thing does not actually solve the problem. (it is just a dirty patch).  The function is may be doing too many things. Just an idea ... if you do not see it, forget it ... may we split this function into more functions, i.e., one per each of the steps, so that we end up with simpler, reusable functions in other scenarios? Note that large functions are hard to reuse.